src/config/spark.py

```diff
@@ -45,17 +44,29 @@ def _base_conf(self):
         return {
             "spark.master": self._get_spark_master(),
             "spark.kubernetes.container.image": KYUUBI_OCI_IMAGE,
-            "spark.kubernetes.authenticate.driver.serviceAccountName": self.service_account,
-            "spark.kubernetes.namespace": self.namespace,
             "spark.submit.deployMode": "cluster",
         }
 
     def _sa_conf(self):
         """Spark configurations read from Spark8t."""
+        if not self.service_account_info:
+            return {}
+
         interface = LightKube(None, None)
         registry = K8sServiceAccountRegistry(interface)
-        if sa := registry.get(f"{self.namespace}:{self.service_account}"):
-            return sa.configurations.props
+
+        account_id = ":".join(
+            [self.service_account_info.namespace, self.service_account_info.service_account]
+        )
+
+        if sa := registry.get(account_id):
+            return {
+                "spark.kubernetes.authenticate.driver.serviceAccountName": self.service_account_info.service_account,
+                "spark.kubernetes.namespace": self.service_account_info.namespace,
```

Comment:
I'm curious, does the `sa.configurations.props` not include these two properties? :thinking:

Reply:
Oh, yes, it is not needed, as it is already provided [here](https://github.com/canonical/spark-k8s-toolkit-py/blob/main/spark8t/domain.py#L322). Great catch!